lib/CL/devices/devices.c

```diff
@@ -431,9 +431,11 @@ pocl_init_devices()
             pocl_devices[dev_index].available = 0;
           }
 
-          if (dev_index == 0)
-            pocl_devices[dev_index].type |= CL_DEVICE_TYPE_DEFAULT;
-
+          if (pocl_get_bool_option ("POCL_WORKAROUND_DEVTYPE", 0) == 0)
```

Comment:
I'd just use POCL_NO_DEFAULT_DEVICE_TYPE_FLAG as its name and be done with it. There might be other bugs related to device-type in the future and we cannot assume this env workarounds them all ;-)